src/porepy/numerics/ad/operators.py

```diff
@@ -1375,22 +1358,20 @@ def __init__(
 
         """
 
-        self._initialize_children()
+        super().__init__(name=name)
+
+    def grid_type(self) -> Literal["subdomains", "interfaces", "boundary grids"]:
```

Comment:
@Yuriyzabegaev @keileg @IvarStefansson 
I propose using the property decorator for such cases, since it is not really a callable and the private return value is static once set.

I marked all of you, since this is a general comment. Might be relevant at several other spots in porepy.

Reply:
I agree

Reply:
I agree, changing it to the property

Reply:
I just noticed that I currently don't use it anywhere outside the class. Do we really need it, or maybe I should remove it completely for brevity?

Reply:
I'd say leave it, could be useful. Perhaps rename to domain_type? @keileg?

Reply:
Agree, let's keep it. And domain_type is a better name.

Reply:
Renamed to domain_type.